NPE in cgeo.geocaching.utils

### Describe your problem!

Reported on Google Play for current versions as well as back to 2021.12.24

```
java.lang.RuntimeException: 
  at android.os.AsyncTask$4.done (AsyncTask.java:415)
  at java.util.concurrent.FutureTask.finishCompletion (FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException (FutureTask.java:252)
  at java.util.concurrent.FutureTask.run (FutureTask.java:271)
  at android.os.AsyncTask$SerialExecutor$1.run (AsyncTask.java:305)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at java.lang.Thread.run (Thread.java:923)
Caused by: java.lang.NullPointerException: 
  at cgeo.geocaching.utils.ImageUtils.lambda$deleteOfflineLogImagesFor$1 (ImageUtils.java:570)
  at cgeo.geocaching.utils.-$$Lambda$ImageUtils$gEAeDrdojSNeedXkStOeXsb0WuY.call (Unknown Source:2)
  at cgeo.geocaching.utils.CollectionStream.map (CollectionStream.java:85)
  at cgeo.geocaching.utils.ImageUtils.deleteOfflineLogImagesFor (ImageUtils.java:570)
  at cgeo.geocaching.ui.ImageListFragment.adjustImagePersistentState (ImageListFragment.java:132)
  at cgeo.geocaching.log.LogCacheActivity$1.doInBackground (LogCacheActivity.java:464)
  at cgeo.geocaching.log.LogCacheActivity$1.doInBackground (LogCacheActivity.java:457)
  at android.os.AsyncTask$3.call (AsyncTask.java:394)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
```


### How to reproduce?

Unclear

### Actual result after these steps?

_No response_

### Expected result after these steps?

_No response_

### Reproducible

Unclear

### c:geo Version

2022.02.16

### System information

```text
Android 10,11,12 - several devices - 77 affected users - 169 occurences
```


### Additional Information

_No response_

--- a/main/src/cgeo/geocaching/utils/ImageUtils.java
+++ b/main/src/cgeo/geocaching/utils/ImageUtils.java
@@ -163,7 +163,7 @@ public final class ImageUtils {
      *            The image format
      * @param quality
      *            The image quality
-     * @param pathOfOutputImage
+     * @param targetUri
      *            Path to store to
      */
     public static void storeBitmap(final Bitmap bitmap, final Bitmap.CompressFormat format, final int quality, final Uri targetUri) {
@@ -567,7 +567,7 @@ public final class ImageUtils {
         if (geocode == null) {
             return;
         }
-        final Set<String> filenamesToKeep = CollectionStream.of(keep).map(i -> i.getFile().getName()).toSet();
+        final Set<String> filenamesToKeep = CollectionStream.of(keep).map(i -> i.getFile() == null ? null : i.getFile().getName()).toSet();
         final String fileNamePraefix = OFFLINE_LOG_IMAGE_PRAEFIX + geocode;
         CollectionStream.of(LocalStorage.getOfflineLogImageDir(geocode).listFiles())
             .filter(f -> f.getName().startsWith(fileNamePraefix) && !filenamesToKeep.contains(f.getName()))

